src/base_alloc/base_alloc_linear.c

```diff
@@ -202,6 +202,8 @@ int umf_ba_linear_free(umf_ba_linear_pool_t *pool, void *ptr) {
             void *ptr = pool + page_size;
             size_t size = pool->metadata.pool_size - page_size;
             ba_os_free(ptr, size);
+            // update pool_size
+            pool->metadata.pool_size = page_size;
```

Comment:
as this is a fix could you add test for that? Where did you find this?

Reply:
I'm afraid it is impossible to add a test for that, because `ba_os_free()` does not return a value. 
It is a very simple and obvious fix. I found it in my head thinking about the code under shower in the evening.
I just got a kind of "Eureka" ;-)

Reply:
@bratpiorka Done